library/core/class.oauth2.php

```diff
@@ -220,13 +222,15 @@ public function setRequestAccessTokenParams($params) {
 
 
     /**
-     * Set additional params to be added to the get string in the getProfile request.
+     * Set additional params to be to be merged with the default parameters
+     * in the getProfile request.
+     *
+     * @param array $params Params to add to the get profile request.
      *
-     * @param string $params.
      * @return $this Return this object for chaining purposes.
      */
-    public function setGetProfileParams($params) {
-        $this->getProfileParams = $params;
+    public function setRequestProfileParams($params = []) {
```

Comment:
If you want to make make sure that calling setRequestProfileParams() will fail you can do:
```php
public function setRequestProfileParams(array $params) {
    $this->requestProfileParams = $params;
    return $this;
}
```